### PR TITLE
refactore(ugov): unify interfaces and mocks

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

* fixes dependency between the `app/inflation` and `x/ugov` (through the import in `expected_keeprs.go`).
* adds generic, reusable mocks to the ugov module

### CODE REVIEW

File: x/ugov/keeper/keeper.go [hunk 1]
```diff
@@ -21,13 +24,23 @@ func NewKeeperBuilder(
...
 	}
 }
 
+// functions to downcast Keeper constructor into super types.
+
+func (kb Builder) Params(ctx *sdk.Context) ugov.ParamsKeeper {
+	return kb.Keeper(ctx)
+}
+func (kb Builder) EmergencyGroup(ctx *sdk.Context) ugov.WithEmergencyGroup {
+	ctx.GasMeter()
```
Comment (gsk967) [inline x/ugov/keeper/keeper.go:40]: is there any reason do add `ctx.GasMeter()` here

File: x/ugov/keeper/keeper.go [hunk 2]
```diff
@@ -21,13 +24,23 @@ func NewKeeperBuilder(
...
 		cdc:   kb.Cdc,
 	}
 }
 
+// functions to downcast Keeper constructor into super types.
+
+func (kb Builder) Params(ctx *sdk.Context) ugov.ParamsKeeper {
+	return kb.Keeper(ctx)
+}
+func (kb Builder) EmergencyGroup(ctx *sdk.Context) ugov.WithEmergencyGroup {
```
Comment (gsk967) [inline x/ugov/keeper/keeper.go:39]:
```suggestion

func (kb Builder) EmergencyGroup(ctx *sdk.Context) ugov.WithEmergencyGroup {
```

File: x/ugov/keeper/keeper.go (hunk 1 above)
Comment (toteki) [inline x/ugov/keeper/keeper.go:40]: Not needed, right?

**Review by gsk967 — APPROVED** [review]
LGTM , I added some comments Please check those

**Review by toteki — APPROVED** [review]

**Pull request merged**